src/toxcore/dht/server.rs

```diff
@@ -692,28 +723,29 @@ mod tests {
         // handle ping response, request from bob peer
         let prs = PingResponsePayload { id: random_u64() };
         let ping_resp = PingResponse::new(&precomp, &bob_pk, prs);
-        let mut client = alice.create_client(&addr, bob_pk);
+        let mut state = alice.state.write();
+        let mut client = alice.create_client(&addr, bob_pk, &state.peers_cache);
         client.ping_id = prs.id;
-        assert!(alice.handle_ping_resp(client, ping_resp).wait().is_ok());
+        assert!(alice.handle_ping_resp(client, ping_resp, &mut state.kbucket).wait().is_ok());
```

Comment:
You should test it by passing special packet and calling `handle_packet`, not by calling handle_YOUR_PACKET directly. See https://github.com/tox-rs/tox/blob/master/src/toxcore/tcp/server/server.rs#L430